session/session.go

```diff
@@ -90,6 +108,152 @@ func Open(config sfdc.Configuration) (*Session, error) {
 	return session, nil
 }
 
+func openDeviceSession(config sfdc.Configuration) (*Session, error) {
+	request, err := buildDeviceAuthenticationFlowInitiationRequest(config.Credentials)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build ")
+	}
+
+	initResp, err := makeDeviceAuthenticationFlowInitiationRequest(request, config.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	var tokenResp *accessTokenResponse
+	var done = make(chan struct{})
+
+	go func() {
+		for {
+			select {
+			case <-time.After(time.Duration(initResp.Interval+1) * time.Second):
+				accessTokenReq, err := buildDeviceAuthenticationFlowAccessRequest(config.Credentials, initResp)
+				if err != nil {
+					log.Fatalf("failed to build device authentication flow access request: %s", err)
```

Comment:
[`log.Fatalf()`](https://golang.org/pkg/log/#Fatalf) will exit the whole program, we should return this error back or maybe `panic` if it's too bad.